### DISCUSSION TITLE
Is there a way to prune xDrip's database to the last N days?

### DISCUSSION THREAD

**Discussion by dv1** [NightscoutFoundation/xDrip#2271] [Q&A]
xDrip has a "Glucose Retention" setting that can be set to let xDrip automatically prune data if it exceeds a certain amount of days. However, I could not find an option to prune already stored data from the database if said data is older than N days. Can this be done in xDrip?

**Comment by Navid200** [comment]
Can you explain when, and why, this will be needed?

**Reply by dv1** [reply]
Well, if I decide _now_ to only retain the last N days worth of data, but I already accumulated a lot more than that, then I need that.

**Reply by Navid200** [reply]
@tolot27 Can you confirm that this happens as described?

**Reply by jamorham** [reply]
Each night xDrip will delete data older than the retention days setting. So if you set it today for 90 days then it will erase everything older than that overnight.

**Reply by dv1** [reply]
Strangely, this did not happen to me. I've set it to 100 days, but I can still see treatments and BG values in the history from February for example. This is version e557cba-2022.05.19 .

**Reply by dv1** [reply]
I now use version e13a76e-2022.08.03 , and the problem persists. Is there no command to explicitly wipe the old data immediately? If not, I have to reinstall xDrip and hope it works then, but I'd like to retain the data from the last 3 months.

**Reply by andrenicGH** [reply]
I'm looking for this feature as well. had data from about a year back, and the app is becoming very slow (I use "libre one minute" data from LinkUpConnect cloud interface). I set Glucose Retention to 90, but the old data is still there.

**Reply by Navid200** [reply]
@andrenicGH How do you know the data is still there?
If you are using the latest stable release, the database cleanup may not run everyday.  It will only run when the phone is being charged.

**Reply by andrenicGH** [reply]
@Navid200 I've checked that the historical BG data can be found under History. I find data there back to when the phone was new (Oct 2023), even with 90 days retention. I also see that the storage space occupied by the app has not changed since I changed retention from 0 to 90.
I've charged my phone at least 3 times with 1h+ duration since I changed this setting, but I'll keep an eye out during the next few days.   
The app is very slow (2-4sec between clicking and response), so I was hoping this could help.
Using latest version I believe, 2024-09-17.
Thanks!

**Reply by Navid200** [reply]
@andrenicGH Well, that's not the latest version.  That's the latest stable release.  Our stable release is updated only twice a year.
But, we have about 50 Nightly updates every year.

Please go to logs: https://navid200.github.io/xDrip/docs/Logs.html
Tap on the magnifying glass symbol.  Search for:
 daily
 
 What do you see?
 
 That will show you when the daily service was run.
 
 If you update to the 2024.11.15 release, you can see the size of the database on the classic status page.  
 
And this shows all the releases and a link for each.  So, you can access any and download and install without changing your update channel.
https://navid200.github.io/xDrip/docs/ReleaseNotes.html

**Reply by andrenicGH** [reply]
@Navid200 , ofc you're right about the versions :)
I see what I assume are the daily jobs scheduled, so I guess that's a good thing.
I tried the "Save Database Daily" option, but turned it off again since it didn't seem to do anything (could not find the files it was supposed to generate). Could be relevant?
You recommend trying the 2024-12-31 version perhaps?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.12.31
 
![image](https://github.com/user-attachments/assets/d6de8433-503d-4191-af41-9fdf99061f9c)

**Reply by Navid200** [reply]
According to those logs, it's not surprising that it has not worked.

First, I suggest updating to 2024.11.15.  Then, go to the classic status page and you can see the database size.  
What is it?  It's pointless to focus on reducing the database size if it is not the cause of the problem to begin with.

If it is large, I will tell you what to do.

**Reply by andrenicGH** [reply]
Thanks, will do!
As of now, Android reports the app uses 300MB for data storage.

**Reply by Navid200** [reply]
Or you can just update to the version that guarantees database cleanup will happen once a day:
2024.12.31

Save database daily occupies storage.  It will not increase the database size and should not slow down xDrip.

**Reply by andrenicGH** [reply]
I updated to https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.12.31 and the daily service is now run. Database size was reduced from 300MB to 60MB and the app is much more responsive! Thanks for the help.
